### PR TITLE
add fall back function in weather download

### PR DESCRIPTION
**Pull request by zhawan**
add fall back function in weather download

### CODE REVIEW

**Review by ArthurJiang — DISMISSED** [review]

File: maro/cli/data_pipeline/base.py [hunk 1]
```diff
@@ -8,11 +8,34 @@
...
 
 from maro.cli.data_pipeline.utils import convert, download_file, StaticParameter, generate_name_with_uuid
+from maro.utils.exception.cli_exception import CommandError
 from maro.utils.logger import CliLogger
 
 logger = CliLogger(name=__name__)
 
 class DataPipeline(ABC):
+    """Base class of data pipeline.
+    Generate scenario/topology specific data for the business engine.
```
Comment (Jinyu-W) [inline maro/cli/data_pipeline/base.py:18]: Insert a blank line above

File: maro/cli/data_pipeline/base.py [hunk 2]
```diff
@@ -8,11 +8,34 @@
...
+    Step 2: Generate the clean data in clean folder.
+    Step 3: Build a binary data file in build folder.
+    The folder structer is:
+    ~/.maro
+            /data/[scenario]/[topology]
+                                    /_download original data file
+                                    /_clean cleaned data file
+                                    /_build bin data file and other necessory files
+                            /meta meta files for data pipeline
+
```
Comment (Jinyu-W) [inline maro/cli/data_pipeline/base.py:30]: too many blank lines

File: maro/cli/data_pipeline/base.py [hunk 3]
```diff
@@ -8,11 +8,34 @@
...
+
+    Args:
+        scenario(str): the scenario of the data.
```
Comment (Jinyu-W) [inline maro/cli/data_pipeline/base.py:33]: Starting with capital letter

File: maro/cli/data_pipeline/base.py [hunk 4]
```diff
@@ -85,68 +91,88 @@ def download(self, is_force: bool):
...
-        Data topology manage multi data pipelines for a specified topology of a research scenario.
-        """
         self._data_pipeline = {}
 
-    def get_build_folders(self):
+    def get_build_folders(self)-> dict:
+        """Get the build file folders of all data pipelines for the topology.
+        
+        Returns:
+            dict : dictionary of build folders, keys are data pipeline names, values
```
Comment (Jinyu-W) [inline maro/cli/data_pipeline/base.py:147]: Is the space after ``dict`` necessary?

File: maro/cli/data_pipeline/citi_bike.py [hunk 5]
```diff
@@ -27,6 +27,23 @@
 
 
 class CitiBikePipeline(DataPipeline):
+    """Generate citi_bike data bin and other necessary files for the specified topology from specified source.
+    They will be generated in ~/.maro/data/citi_bike/[topology]/_build.
```
Comment (Jinyu-W) [inline maro/cli/data_pipeline/citi_bike.py:31]: Same to the comments above: blank line below the summary line; start each arg with capital letter....

Same to other docstring, please check.

**Review by Jinyu-W — CHANGES_REQUESTED** [review]

File: maro/cli/data_pipeline/citi_bike.py [hunk 6]
```diff
@@ -27,6 +27,24 @@
...
+            /data/citi_bike/[topology]
+                                    /_build bin data file and other necessory files
+                                    /source
+                                            /_download original data files
+                                            /_clean cleaned data files
+            /temp download temp files
+
+    Args:
+        topology(str): Topology name of the data files
+        source(str): Source url of original data file
```
Comment (Jinyu-W) [inline maro/cli/data_pipeline/citi_bike.py:44]: end with period.

File: maro/cli/data_pipeline/citi_bike.py [hunk 7]
```diff
@@ -240,18 +240,35 @@ def _preprocess(self, unzipped_file: str):
...
+                                    /_build bin data file
+                                    /source
+                                            /_download original data file
+                                            /_clean cleaned data file
+            /temp download temp file
+
+    Args:
+        topology(str): Topology name of the data file
+        source(str): Source url of original data file
+        is_temp(bool): (optional) If the data file is temporary
```
Comment (Jinyu-W) [inline maro/cli/data_pipeline/citi_bike.py:270]: End with period.

File: maro/cli/data_pipeline/citi_bike.py [hunk 8]
```diff
@@ -351,16 +352,16 @@ def _preprocess(self, input_file: str, output_file: str):
...
-        trip_source(str): original source url of citi_bike data 
-        station_info(str): current status station info of the stations
-        weather_source(str): original source url of weather data
-        is_temp(bool): (optional) if the data file is temporary
+        topology(str): Topology name of the data file
+        trip_source(str): Original source url of citi_bike data 
+        station_info(str): Current status station info of the stations
+        weather_source(str): Original source url of weather data
+        is_temp(bool): (optional) If the data file is temporary
     """
```
Comment (Jinyu-W) [inline maro/cli/data_pipeline/citi_bike.py:363]: End with period.

**Review by Jinyu-W — CHANGES_REQUESTED** [review]

**Review by Jinyu-W — APPROVED** [review]
* check the test status of this PR

**Pull request merged**